### PR TITLE
Add functions to auth to all projects at SP

### PR DESCRIPTION
**Pull request by gsilvis**

### CODE REVIEW

**Comment by gsilvis** [comment]
Still needs to be tested on a real setup, to make sure there isn't something dumb

File: mixmatch/auth.py [hunk 1]
```diff
@@ -53,26 +55,45 @@ def get_local_auth(user_token):
...
+        local_auth,
+        service_provider
+    )
+    return session.Session(auth=remote_auth)
+
+
+def get_projects_at_sp(service_provider, user_token):
+    """Perform K2K auth, and return the projects that can be scoped to."""
+    config = get_conf_for_sp(service_provider)
+    session = get_unscoped_sp_auth(service_provider, user_token)
```
Comment (knikolla) [inline mixmatch/auth.py]: don't shadow global vars

File: mixmatch/auth.py [hunk 2]
```diff
@@ -53,26 +55,45 @@ def get_local_auth(user_token):
...
+    headers = {
```
Comment (knikolla) [inline mixmatch/auth.py]: https://github.com/openstack/keystoneauth/blob/master/keystoneauth1/session.py#L711

**Comment by gsilvis** [comment]
This has now been tested on a real setup.  `openstack image list`, and `openstack image show` on both local and remote images all work.

**Comment by knikolla** [comment]
+1, merging.

**Pull request merged**